lib/compilers/vb.js

```diff
@@ -0,0 +1,159 @@
+        const coreClrJitPath = path.join(clrBuildDir, this.compilerProps('compiler.vb.jit'));
+        const crossgen2Path = path.join(clrBuildDir, 'crossgen2', this.compilerProps('compiler.vb.crossgen2'));
+
+        const programPublishPath = path.join(
+            programDir,
+            'bin',
+            this.buildConfig,
+            this.targetFramework,
+            this.rID,
+            'publish',
+        );
+
+        const programDllPath = path.join(programPublishPath, 'CompilerExplorer.dll');
+
+        const projectFileContent =
+            `<Project Sdk="Microsoft.NET.Sdk">
+            <PropertyGroup>
+                <TargetFramework>${this.targetFramework}</TargetFramework>
+                <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+                <Nullable>enable</Nullable>
+                <AssemblyName>CompilerExplorer</AssemblyName>
+            </PropertyGroup>
+         </Project>
+        `;
+
+        execOptions.customCwd = programDir;
+        await fs.writeFile(projectFilePath, projectFileContent);
+        options = this.compilerOptions;
+
+        const compilerResult = await super.runCompiler(
+            compiler,
+            options,
+            inputFileName,
+            execOptions,
+        );
+
+        if (compilerResult.code !== 0)
+            return compilerResult;
+
+        const crossgen2Result = await this.runCrossgen2(
+            execOptions,
+            crossgen2Path,
+            coreClrJitPath,
+            programPublishPath,
+            programDllPath,
+            this.getOutputFilename(programDir, ''),
+        );
+
+        if (crossgen2Result.code !== 0)
+            return crossgen2Result;
+
+        return compilerResult;
+    }
+
+    optionsForFilter() {
+        return this.compilerOptions;
+    }
+
+    getOutputFilename(dirPath) {
+        return path.join(dirPath, `output.s`);
+    }
+
+    cleanAsm(asm) {
+        let cleanedAsm = '';
+        const asmLines = asm.replace(/\r/g, '').split('\n');
+
+        for (const line of asmLines) {
+            if (line.startsWith('; Assembly listing for method')) {
+                // ; Assembly listing for method ConsoleApplication.Program:Main(System.String[])
+                //                               ^ This character is the 31st character in this string.
+                // `substring` removes the first 30 characters from it and uses the rest as a label.
+                cleanedAsm = cleanedAsm.concat(line.substring(30) + ':\n');
+                continue;
+            }
+
+            if (line.startsWith(';') ||
+                line.startsWith('Emitting R2R PE file')) {
+                continue;
+            }
+
+            cleanedAsm = cleanedAsm.concat(line + '\n');
+        }
+
+        return cleanedAsm;
+    }
+
+    async runCrossgen2(execOptions, crossgen2Path, jitPath, publishPath, dllPath, outputPath) {
+        const targetOS = this.compilerProps('compiler.vb.targetOS');
```

Comment:
There's a lot of overlap/copy/pasting here, perhaps it's better to make a base function of sorts that does this and uses 'vb'/'csharp'/'fsharp' as a compilerKeyId variable in the class or use `this.key()` which seems to be the same.

Reply:
Dome. I'm using `this.lang.id` instead.

Reply:
Cool, smart